src/generic/listctrl.cpp

```diff
@@ -2197,6 +2203,23 @@ void wxListMainWindow::HighlightAll( bool on )
     }
 }
 
+bool wxListMainWindow::HighlightOnly( size_t line )
+{
+    if ( IsEmpty() || line == (size_t)-1 )
+        return false;
+
+    for ( size_t l = 0; l <= GetItemCount() - 1; l++ )
+    {
+        if ( l != line && HighlightLine(l, false) )
+            RefreshLine(l);
+    }
```

Comment:
```suggestion
    HighlightLines(0, GetItemCount() - 1, false);
```
This is both simpler and much more efficient for the virtual list controls.

Reply:
Yes it is, but we don't want to unselect the current line and select it again afterwards. this is redundant and will result in a spurious `wxEVT_LIST_ITEM_SELECTED` and `wxEVT_LIST_ITEM_DESELECTED` events. this is also the case if we try to activate an already selected item, and `wxMSW` doesn't behave like that!

Either way, this is an optimisation and sure we can come up with a clever why to optimise this in the future.

Reply:
Done here 67022b4

Reply:
Still need optimisation for non virtual list controls though!

Reply:
@vadz  Is there a technical reason not to use `wxSelectionStore` for all list controls by default (not just virtual LCs)? or just switching to using `wxSelectionStore` for them requires a lot of work and changes that might not be as easy as it appears?

Reply:
This code predates `wxSelectionStore`, so there is the issue of having to update it to use it, and risking introducing bugs while doing it... but I think it would be indeed worthwhile to do it -- but just not as part of this PR, which I'll try to review and merge a.s.a.p., before it becomes any bigger.